docs/tutorials/algokit-template.md

```diff
@@ -0,0 +1,143 @@
+# Creating AlgoKit Templates
+
+This README serves as a guide on how to create custom templates for AlgoKit, a tool for initializing Algorand smart contract projects. 
+Creating templates in AlgoKit involves the use of various configuration files and a templating engine to generate project structures that are tailored to your needs.
+This guide will cover the key concepts and best practices for creating templates in AlgoKit.
+We will also refer to the official [`algokit-beaker-default-template`](https://github.com/algorandfoundation/algokit-beaker-default-template) as an example.
+## Table of Contents
+
+- [Overview of AlgoKit Templates](#overview-of-algokit-templates)
+  - [Copier/Jinja](#copierjinja)
+  - [AlgoKit Functionality with Templates](#algokit-functionality-with-templates)
+- [Key Concepts](#key-concepts)
+  - [.algokit.toml](#algokittoml)
+  - [Python Support: pyproject.toml](#python-support-pyprojecttoml)
+  - [TypeScript Support: package.json](#typescript-support-packagejson)
+  - [Bootstrap Option](#bootstrap-option)
+  - [Template Answers and Variables](#template-answers-and-variables)
+- [Recommendations](#recommendations)
+- [Conclusion](#conclusion)
+
```

Comment:
Let's add another section called quick start in here before any other sections, because despite overview of the templates itself we still don't directly provide a guide on how exactly to create them for users. So it will serve a quick TL;DR section that users runs over before he dives into reading the tutorial because he will need the reference beaker template anyway.

Since we are assuming beaker templates lets do a simple set of numbered items on things that a template builder needs to setup before he reads the remaining part of this tutorial. 

Something like:

```
For users who are keen on getting started with creating AlgoKit templates, you can follow these quick steps:

1. Click on `Use this template`->`Create a new repository` on [algokit-beaker-default-template](https://github.com/algorandfoundation/algokit-beaker-default-template) Github page. This will create a new reference repository with clean git history, allowing you to start modifying and transforming the base beaker template into your own custom template. 
2. Modify the cloned template according to your specific needs. You can refer to the remainder of this tutorial for an understanding of expected behaviors from the AlgoKit side, Copier - the templating framework, and key concepts related to the default files you will encounter in the reference template.
```

@Loedn please note merging this PR assumes we turned the repos to template repos (at least the beaker repository) so this ways folks wouldn't need to Clone-> Clear git history each time but just make a templage repo from base template repo.